src/main/java/org/opensearch/security/configuration/DlsFlsValveImpl.java

```diff
@@ -407,6 +407,15 @@ public void handleSearchContext(SearchContext searchContext, ThreadPool threadPo
                 }
             }
 
+            // Restrict access for star tree index if there are any DLS/FLS/field masking restrictions
+            if (searchContext.getQueryShardContext().getStarTreeQueryContext() != null) {
+                FieldPrivileges.FlsRule fls = config.getFieldPrivileges().getRestriction(privilegesEvaluationContext, index);
+                FieldMasking.FieldMaskingRule rule = config.getFieldMasking().getRestriction(privilegesEvaluationContext, index);
+                if (!dlsRestriction.isUnrestricted() || !fls.isUnrestricted() || !rule.isUnrestricted()) {
```

Comment:
Nit: As you only need the information whether something is restricted or not, you can also use the method `isUnrestricted()`: https://github.com/opensearch-project/security/blob/7262ec00696886e22e81a26e6e2a1450e2de8bcd/src/main/java/org/opensearch/security/privileges/dlsfls/AbstractRuleBasedPrivileges.java#L188

As this won't need to collect rules, this might have a small performance advantage.

Reply:
Thanks have done this